### PR TITLE
Gender differences sprites, part 1

### PR DESCRIPTION
**Pull request by SubzeroEclipse**
## Description
First PR that adds sprites for some Pokémon with gender based differences, namely:
- Pikachu/Raichu
- Gloom/Vileplume
- Doduo/Dodrio
- Goldeen/Seaking
- Xatu
- Politoed
- Meditite/Medicham
- Gible/Gabite/Garchomp

Pikachu and Gloom's back sprites were taken from [Chaos Rush's resource](https://www.pokecommunity.com/showthread.php?t=267728); Pikachu's front was done by CyanSixFour.

## **Discord contact info**
SubzeroEclipse#2104

### CODE REVIEW

File: src/data/pokemon/species_info.h [hunk 1]
```diff
@@ -36,6 +36,7 @@
         .abilities = {ABILITY_STATIC, ABILITY_NONE, ABILITY_LIGHTNING_ROD},\
         .bodyColor = BODY_COLOR_YELLOW,                                    \
         .noFlip = flip,                                                    \
+        .flags = SPECIES_FLAG_GENDER_DIFFERENCE,                           \
```
Comment (AsparagusEduardo) [inline src/data/pokemon/species_info.h:39]: Adding this in the macro will cause *all* Pikachu forms to be counted as having gender differences, which is not the case.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Comment by SubzeroEclipse** [comment]
Fixed and ready for review

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
Looks good in-game, though #3051 made us realize that bit depths for some sprites were wrong.
You'll need to solve conflicts and fix the bit depths of the following sprites due to them being copied from the erroneous previous depths:
- Front Female Dodrio
- Front Female Xatu

**Comment by SubzeroEclipse** [comment]
Ready for re-review! Fixed conflicts and Dodrio/Xatu sprites and swapped sprites as discussed on Discord

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**